web_src/js/markup/anchors.js

```diff
@@ -2,10 +2,10 @@ import {svg} from '../svg.js';
 
 const headingSelector = '.markup h1, .markup h2, .markup h3, .markup h4, .markup h5, .markup h6';
 
-function scrollToAnchor() {
-  if (document.querySelector(':target')) return;
-  if (!window.location.hash || window.location.hash.length <= 1) return;
-  const id = decodeURIComponent(window.location.hash.substring(1));
+function scrollToAnchor(hash) {
+  if (document.querySelector(':target')) return; // something else on the page matches
+  if (hash?.length <= 1) return;
```

Comment:
```suggestion
  if (!hash || document.querySelector(':target')) return; // no hash given or something else on the page matches
```
Also, isn't the second condition a bug?
Why would you not scroll to another anchor if something else is already targeted through the URL?
I simply don't see where it would be useful…

Reply:
1. Our `id`s in HTML never match the anchors, so if there is a match we know that something else on the page has matchd and we generally do not want to interfere with the browser-native behaviour.
2. I kept it on separate line so the comment association works
3. `!hash` introduces a bug when hash is just `#`.

Reply:
I can understand 2. and 3, but isn't 1. exactly the same case as described in the related issue?

Reply:
see for example githubs behavior of highlighting the box of the file your URL points at.
I'm fairly certain that it would still scroll, despite highlighting something else already?

Reply:
Take this HTML:

```
<h2 id="user-content-configuration">
  <a class="anchor" href="#configuration">svg</a>
  Configuration
</h2>
```

Clicking the link triggers js to scroll to `user-content-configuration`. I don't know the exact reason for this mismatch but it may be security-related.

Now the spec for `:target`:

> The :target [CSS](https://developer.mozilla.org/en-US/docs/Web/CSS) [pseudo-class](https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-classes) represents a unique element (the target element) with an [id](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes#id) matching the URL's fragment.

With our markdown anchors, `:target` can never exist, and if it does, another anchor outside markdown is active, take for example hypothetical `#toc` that would link to a TOC outside. If the same name exists as `user-content-toc`, we would scroll to the markdown anchor without this condition. I generally think our own IDs are more important than ones created in user content, so I give our own a preference in such a conflict.

Reply:
The point you are missing may be that if `:target` exists, the browser-has already scrolled somewhere (without JS), so if we remove the condition, it may scroll twice, once natively, and once via JS.

Reply:
I guess a point could be made for subsequent user-initiated click on conflicting anchors in markdown, but I'm not sure how we could detect such cases. I think they are currently non-existant/very uncommon.

Reply:
Fixed below, it was actually a very easy fix.